torch/_inductor/simple_fsdp/utils.py

```diff
@@ -52,32 +52,59 @@ def compute_node_users(
 
     return inverse_users, node_users
 
+def _check_ir_node_fsdp(ir_node: "ir.Operation") -> bool:
+    """
+    Determine if the AG/RS node is for FSDP or TP
+    """
+    if config.simplefsdp.tp_world_size < 0:
+        return True
+
+    is_fsdp = False
+    ir_node_origins = list(getattr(ir_node, "origins", None))
+
+    if len(ir_node_origins) == 0:
+        # bucketed AG and RS doesn't have origins, but they are created by FSDP
+        is_fsdp = True
+
+    for n in ir_node_origins:
+        meta_data = n.meta.get("stack_trace", {})  
+        # TODO(ruisizhang123): hack to get FSDP node (the FSDP AG/RS are created from torch_spmd)
+        if 'torch_spmd' in meta_data:
```

Comment:
maybe let's use `parametrization` as a more fundamental characterization of when FSDP happens. We may need to revisit it if later we want to also enable DDP/HSDP.